augur/utils.py

```diff
@@ -453,3 +455,74 @@ def nthreads_value(value):
         return int(value)
     except ValueError:
         raise argparse.ArgumentTypeError("'%s' is not an integer or the word 'auto'" % value) from None
+
+
+def all_parents(tree):
+    '''
+    Return dictionary mapping child node names to parent node names
+    '''
+    parents = {}
+    for clade in tree.find_clades(order='level'):
+        for child in clade:
+            parents[child.name] = clade.name
+    return parents
+
+
+def annotate_parents(tree):
+    # Get all parent nodes by node.
+    parents_by_node = all_parents(tree)
+
+    # Next, annotate each node with its parent.
+    for node in tree.find_clades():
+        if node == tree.root:
+            node.up = None
```

Comment:
I'd think that `node.parent` would be the more usual naming, but could be wrong.